# --- SDE-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
#
# Filename: package/.../pcmciautils/udev-rules-hotfix.patch
# Copyright (C) 2007 The OpenSDE Project
#
# More information can be found in the files COPYING and README.
#
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- SDE-COPYRIGHT-NOTE-END ---

diff -ruN pcmciautils-014.orig/udev/rules-base pcmciautils-014/udev/rules-base
--- pcmciautils-014.orig/udev/rules-base	2006-06-01 11:07:52.000000000 +0200
+++ pcmciautils-014/udev/rules-base	2007-01-08 21:30:31.000000000 +0100
@@ -3,12 +3,12 @@
 # are so broken that we need to read out random bytes of it
 # instead of the manufactor, card or product ID. Then the
 # matching is done in userspace.
-ACTION=="add", SUBSYSTEM=="pcmcia", MODALIAS=="?*", \
+ACTION=="add", SUBSYSTEM=="pcmcia", ENV{MODALIAS}=="?*", \
 		RUN+="/sbin/pcmcia-check-broken-cis"
 
 # However, the "weak" matching by func_id is only allowed _after_ modprobe
 # returns, so that "strong" matches have a higher priority.
-ACTION=="add", SUBSYSTEM=="pcmcia", MODALIAS=="?*", \
+ACTION=="add", SUBSYSTEM=="pcmcia", ENV{MODALIAS}=="?*", \
 		RUN+="/bin/sh -c 'echo 1 > /sys/$devpath/allow_func_id_match'"
 
 # PCMCIA sockets:
diff -ruN pcmciautils-014.orig/udev/rules-modprobe pcmciautils-014/udev/rules-modprobe
--- pcmciautils-014.orig/udev/rules-modprobe	2006-06-01 11:07:52.000000000 +0200
+++ pcmciautils-014/udev/rules-modprobe	2007-01-08 21:30:31.000000000 +0100
@@ -1,3 +1,3 @@
 # modprobe $modalias loads all possibly appropriate modules
-ACTION=="add", SUBSYSTEM=="pcmcia", MODALIAS=="?*", \
-		RUN+="/sbin/modprobe $modalias"
+ACTION=="add", SUBSYSTEM=="pcmcia", ENV{MODALIAS}=="?*", \
+		RUN+="/sbin/modprobe $env{MODALIAS}"
